### PR TITLE
Modernize plugin

### PR DESCRIPTION
**Pull request by daniel-beck**
* "Jenkins" and "Plugin" get stripped from the name, so remove that and capitalize the rest.
* Update to current parent POM
* Address FindBugs warnings from new parent POM
* Update core and Java dependency to not require a dozen unused implied dependencies: https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/ClassicPluginStrategy.java#L413...L427
* Remove empty Javadoc comments that just clog up the tubes.
* Add null-ness annotations

----

Tests still pass, otherwise I'm not confident I didn't mess up some of the internals. Review is needed.

### CODE REVIEW

**Comment by daniel-beck** [comment]
If you're concerned about the core dependency increase:
http://stats.jenkins.io/pluginversions/build-timeout.html
1.18 is 10 months old (plenty of time to update) and 100% of its users are on that core version or newer (96% of all users of all versions of the plugin are).

TBH if it were my plugin, I'd go with a 2.x release, but the main point here is to get rid of the unused implied dependencies to detached plugins.

**Comment by ikedam** [comment]
Upgrading the core looks reasonable to me. Targetting old cores cause installing unnecessary plugins as you point, and build-timeout is often installed during the setup wizard.
(At the same time, build-timeout doesn't support pipeline, and may not be useful for many people)

I wonder why ci.jenkins.io doesn't test this request even after merging #65 ...

File: pom.xml [hunk 1]
```diff
@@ -77,13 +89,6 @@
...
         <plugins>
             <plugin>
-                <artifactId>maven-surefire-plugin</artifactId>
-                <configuration>
-                    <!-- jenkins-test-harness < 1.545 doesn't support concurrent tests. -->
-                    <forkCount>1</forkCount>
-                </configuration>
-            </plugin>
-            <plugin>
                 <artifactId>maven-enforcer-plugin</artifactId>
```
Comment (ikedam) [inline pom.xml]: We can remove also configurations for maven-enforcer-plugin.

**Comment by ikedam** [comment]
Test failures caused for
* The test expects the log line "Test" from `FakeBuildStep`.
* Tests on windows always log following output:
```
Building in workspace C:\Users\ikedam\AppData\Local\Temp\jenkinsTests.tmp\jenkins1377785768819663524test\workspace\test0
```

So you can fix the test failures by modifying the log line from `FakeBuildStep` to more appropriate identifiable string.

And we can replace `BuildStepWithTimeoutTest#assertLogDoesNotContain` to `JenkinsRule#assertLogNotContains`, which makes test errors more descriptive.

**Comment by daniel-beck** [comment]
@ikedam I addressed your review comments.

**Comment by ikedam** [comment]
Released build-timeout-1.19.
It allows users install the plugin without installing matrix-project so on.

**Comment by daniel-beck** [comment]
@ikedam Thanks!

File: pom.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
   
   <artifactId>build-timeout</artifactId>
   <packaging>hpi</packaging>
-  <name>Jenkins build timeout plugin</name>
+  <name>Build Timeout</name>
```
[the remark this answers is not in the document]
Comment (daniel-beck) [reply pom.xml]: Relying on the display name that only exists for human use is a terrible idea. Scripts should rely on the immutable artifactId instead.

**Pull request merged**